### ISSUE TITLE
load_macros_from_path setting doesn't seem to be loaded from pyproject.toml

### ISSUE THREAD

**Issue by imrehg** [sqlfluff/sqlfluff#3896]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Using `sqlfluff` with a `dbt` project, running with the `jinja` templater. I was trying to put all the relevant `sqlfluff` config into `pyproject.toml` as that is a good [ultimate config](https://docs.sqlfluff.com/en/latest/configuration.html#configuration) (values from there override anything else). I nevertheless run into jinja/macro related issues, and seemed like sqlfuff doesn't pick up some configuration from that file, specifically the `load_macros_from_path`. On the other hand, if the same config line is added into `.sqlfluff`, it works like a charm.

### Expected Behaviour

I expected this configuration (non-relevant parts omitted)

```
# pyproject.toml
[tools.sqlfluff.templater.jinja]
load_macros_from_path = "macros"
```

to have equivalent effect as this one:

```
# .sqlfluff
[sqlfluff:templater:jinja]
load_macros_from_path = macros
```

### Observed Behaviour

Using config of

```
# pyproject.toml
[tool.sqlfluff.core]
templater = "jinja"

[tools.sqlfluff.templater.jinja]
apply_dbt_builtins = true
load_macros_from_path = "macros"
```

and running 

```shell
sqlfluff lint -vv ...
```

I see

```
[snip]
== Raw Config:
[snip]
templater:                              
    jinja:                                  
        apply_dbt_builtins: True

==== readout ====
...
```

If I create a `.sqlfluff` with just this:

```
# .sqlfluff
[sqlfluff:templater:jinja]
load_macros_from_path = macros
```

(provided that macros folder exists) then the run results in

```
[snip]
== Raw Config:
[snip]
templater:                              
    jinja:                                  
        apply_dbt_builtins: True
        load_macros_from_path:/home/imrehg/test/macros

==== readout ====
...
```

and things working.

### How to reproduce

In an empty folder, I guess

```shell
mkdir macros

cat > macros/echo.sql <<EOF
{% macro echo(column_name) %}
    {{ column_name }}
{% endmacro %}
touch empty.sql
EOF

cat > test.sql <<EOF
SELECT
{{ echo('*') }}
FROM source;
EOF

cat > pyproject.toml <<EOF
[tool.sqlfluff.core]
templater = "jinja"

[tools.sqlfluff.templater.jinja]
apply_dbt_builtins = true
load_macros_from_path = "macros"
EOF

sqlfluff lint -vv --dialect ansi test.sql
```

This will fail and won't show the `load_macros_from_path` in the config

```
cat > .sqlfluff <<EOF                                                                         
[sqlfluff:templater:jinja]
load_macros_from_path = macros
EOF

sqlfluff lint -vv --dialect ansi test.sql
```

Will succeed and show the setting.

### Dialect

N/A

### Version

```
sqlfluff, version 1.3.2
Python 3.9.12
```

### Configuration

Variable, as mentioned above.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by imrehg** [comment]
Closing this report with total prejudice, as it's my typo of using `tools` rather than `tool` in `pyproject.toml`.

Good reminder that "an hour debugging can save you a few seconds of proofreading". But I did pick up some sqlfluff internals while doing it, so there's that.
